components/profile_popover/index.js

```diff
@@ -39,15 +39,19 @@ function mapStateToProps(state, ownProps) {
     }
 
     const selectedPost = getSelectedPost(state);
-    let channelId;
-    if (selectedPost.exists === false) {
-        const currentChannel = getCurrentChannel(state) || {};
-        channelId = currentChannel.id;
-    } else {
-        channelId = selectedPost.channel_id;
+
+    let channelId = ownProps.channelId;
+
+    if (!channelId) {
```

Comment:
Is it feasible to eliminate this and expect `channelId` to always be supplied?

Reply:
`AtMention` was the hold up. Took a bit, but I think we should be good now without this logic. Thanks for catching—an upstream refactor I just committed should prevent a situation in GlobalThreads ThreadViewer where the channelId to `ProfilePopover` (of an `AtMention`) would've been undefined (resulting in a soft fail and just wouldn't display the 'Channel Admin' badge. Really minor but good to catch now.